src/main/java/com/autotune/common/utils/ExponentialBackOff.java

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * This builder class helps in retrying specific task using exponential backoff approach.
+ */
+public final class ExponentialBackOff {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExponentialBackOff.class);
+    // Maximum total time in Millis to complete exponential backoff approach. Default set to 10sec
+    public static long defaultMaxElapsedTimeMillis = 10 * 1000;   //1o Seconds
+    // Default time to wait for each retry
+    public static long defaultTimeToWait = 1000;
+    // increases the back off period for each retry attempt using a randomization function that grows exponentially.
+    // randomized_interval = retryIntervalMillis * (random value in range [1 - randomization_factor, 1 + randomization_factor])
+    public static double defaultRandomizationFactor = 0.5;
+    // Some task need initial sleep before proceedings . defaults to zero.
+    public static long defaultInitialIntervalMillis = 0;
+    //  increases the back off period for each retry attempt using a randomization function that grows exponentially.
+    public static double defaultMultiplier = 0.5;
+    // Set retry after backoff exhausted with MaxElapsedTimeMillis , default is set to zero that means no retries.
+    private final int numberOfRetries;
+    // Some task need initial sleep before proceedings .
+    private final long initialIntervalMillis;
+    // Maximum total time in Millis to complete exponential backoff approach.
+    private final long maxElapsedTimeMillis;
+    private final Random random = new Random();
+    //  increases the back off period for each retry attempt using a randomization function that grows exponentially.
+    private final double randomizationFactor;
+    private final double multiplier;
+    private int numberOfTriesLeft;
+    private long retryIntervalMillis;
+    private long randomizedIntervalMillis;
+    private double totalRetryIntervalMillis = 0;
+
+
+    public ExponentialBackOff(Builder builder) {
+        this.numberOfRetries = builder.numberOfRetries;
+        this.numberOfTriesLeft = builder.numberOfRetries;
+        this.initialIntervalMillis = (builder.initialIntervalMillis > 0) ? builder.initialIntervalMillis : defaultInitialIntervalMillis;
+        this.maxElapsedTimeMillis = (builder.maxElapsedTimeMillis > 0) ? builder.maxElapsedTimeMillis : defaultMaxElapsedTimeMillis;
+        this.randomizationFactor = (builder.randomizationFactor > 0) ? builder.randomizationFactor : defaultRandomizationFactor;
+        this.retryIntervalMillis = (builder.retryIntervalMillis > 0) ? builder.retryIntervalMillis : defaultTimeToWait;
+        this.multiplier = (builder.multiplier > 0) ? builder.multiplier : defaultMultiplier;
+    }
+
+    public boolean shouldRetry() {
+        boolean shouldRetry = false;
+        if (Math.round(this.totalRetryIntervalMillis) < this.maxElapsedTimeMillis)
+            shouldRetry = true;
+        else if (numberOfRetries > 0) {
+            if (numberOfTriesLeft > 0) {
```

Comment:
Can the both if conditions be merged into one? Something like `else if (numberOfRetries > 0 && numberOfTriesLeft > 0)`?